src/sagemaker/serve/builder/model_builder.py

```diff
@@ -543,15 +571,16 @@ def build(
         self,
         mode: Type[Mode] = None,
         role_arn: str = None,
-        sagemaker_session: str = None,
+        sagemaker_session: Optional[Session] = None,
```

Comment:
Isn't it a breaking change?

Reply:
Why though? Python doesn't enforce type hints, and it's used as Session object down the line.

Reply:
yes, but I am concern on the function contract change.

If a customer has build(session=str) already, wouldn't the pip update cause their code to fail, which then forces them to either stop the update or fix the code. While the latter is better, unless it is essential, why would we want to introduce such pain to the customer?

Reply:
Please note, I am inclined that session here should not be str. 

However, SDK release is mostly minor or patch version (as of today) which aim to maintain backward compatibility. However, this change isn't.

Reply:
hmm, I don't think if customers pass in a string as a sagemaker_session, it will actually work. If their code work already, they must have already passed in a session object not a str.